cmd/options/node_options.go

```diff
@@ -21,6 +21,15 @@ import (
 )
 
 // NodeOptions contains options and configuration settings for the node service.
-type NodeOptions struct{}
+type NodeOptions struct {
+	// VolumeAttachLimitOverwrite specifies the value that shall be reported as "maximum number of attachable volumes"
+	// in CSINode objects. It is similar to https://kubernetes.io/docs/concepts/storage/storage-limits/#custom-limits
+	// which allowed administrators to specify custom volume limits. Note that this solution is just a "stop-gap"
+	// solution until a more sophisticated solution can be implemented (dynamically discovering the maximum number of
+	// attachable volume per EC2 machine type, see also https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/347).
+	VolumeAttachLimitOverwrite int64
+}
 
-func (s *NodeOptions) AddFlags(fs *flag.FlagSet) {}
+func (o *NodeOptions) AddFlags(fs *flag.FlagSet) {
+	fs.Int64Var(&o.VolumeAttachLimitOverwrite, "volume-attach-limit-overwrite", -1, "Overwrite value for the maximum number of attachable volumes for all nodes")
```

Comment:
I would prefer this is simply called `volume-attach-limit` integer, so as if not specified the documentation will say - value is derived from instance type.